### PR TITLE
[iOS] Fix SearchBar focus issues 

### LINKED ISSUE #11542: SearchBar does not unfocus and hide keyboard on iOS
### Description

I'm trying to create a search bar with some results in a list view that appear. When a user taps a result in the list view or presses search in the keyboard I try to unfocus the search bar to hide the keyboard, but the keyboard does not disappear even though the event fires and calls unfocus for the search bar.

### Steps to Reproduce

1. Create a new content page
2. Add a searchbar and a list view
3. Add event handler that unfocuses searchbar when a listview item is tapped
4. Keyboard will not disappear

### Link to public reproduction project repository

https://github.com/3steve3/SearchBarBug

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 16

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Fix SearchBar focus issues on iOS/Catalyst.

fix-searchbar-focus-ios

To test/validate the changes, launch the .NET MAUI Gallery and navigate to the SearchBar sample. Tap on the Focus SearchBar. A dialog must appear. 

### Issues Fixed

Fixes #11542

### CODE REVIEW

**Review by PureWeen — CHANGES_REQUESTED** [review]
What do you think about my comment here?
https://github.com/dotnet/maui/issues/11542#issuecomment-1323965626

What if we if we fixed the "UnFocus" API so users could control if they want to open or close the keyboard?

**Comment by jsuarezruiz** [comment]
> Reference in



> What do you think about my comment here? [#11542 (comment)](https://github.com/dotnet/maui/issues/11542#issuecomment-1323965626)
> 
> What if we if we fixed the "UnFocus" API so users could control if they want to open or close the keyboard?

Do you mind to move the logic to ViewHandler and check always the PlatformView status?

**Review by mandel-macaque — APPROVED** [review]
minor changes.

File: src/Core/src/Handlers/SearchBar/SearchBarHandler.iOS.cs [hunk 1]
```diff
@@ -187,6 +197,12 @@ void OnEditingChanged(object? sender, EventArgs e)
 			UpdateCancelButtonVisibility();
 		}
 
+		void OnEditingStopped(object? sender, EventArgs e)
+		{
+			if (VirtualView != null)
```
[the remark this answers is not in the document]
Comment (jsuarezruiz) [reply src/Core/src/Handlers/SearchBar/SearchBarHandler.iOS.cs]: Changed.

**Review by PureWeen — CHANGES_REQUESTED** [review]
Can you add tests?

**Comment by jsuarezruiz** [comment]
> Can you add tests?

Added Device Test, could you review it again?

File: src/Core/tests/DeviceTests/Handlers/SearchBar/SearchBarHandlerTests.iOS.cs [hunk 2]
```diff
@@ -135,6 +136,48 @@ public async Task CharacterSpacingInitializesCorrectly()
 			Assert.Equal(xplatCharacterSpacing, values.PlatformViewValue);
 		}
 
+		[Fact]
+		public async Task SearchBarFocusedIsWorking()
```
Comment (PureWeen) [inline src/Core/tests/DeviceTests/Handlers/SearchBar/SearchBarHandlerTests.iOS.cs:140]: Can we run the code through the handler and simplify this a bit to 

```C#
[Fact]
public async Task SearchBarFocusedIsWorking()
{
	var searchBar1 = new SearchBarStub
	{
		Text = "SearchBar 1"
	};

	await InvokeOnMainThreadAsync(async () =>
	{
		var contentViewHandler = CreateHandler(searchBar1);
		await contentViewHandler.PlatformView.AttachAndRun(async () =>
		{
			searchBar1.Handler.Invoke(nameof(IView.Focus), new FocusRequest(false));
			await searchBar1.WaitForFocused();
			Assert.True(searchBar1.IsFocused);

			searchBar1.Handler.Invoke(nameof(IView.Unfocus), new FocusRequest(false));
			await searchBar1.WaitForUnFocused();
			Assert.False(searchBar1.IsFocused);
		});
	});
}
```

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Core/tests/DeviceTests/Handlers/SearchBar/SearchBarHandlerTests.iOS.cs (hunk 2 above)
Comment (jsuarezruiz) [reply src/Core/tests/DeviceTests/Handlers/SearchBar/SearchBarHandlerTests.iOS.cs:140]: Added.

**Review by PureWeen — APPROVED** [review]

**Pull request merged**